src/serializers/decoders/prev/v10/decode_graph.c

```diff
@@ -0,0 +1,218 @@
+	 *  Payload(s) X N
+	 * */
+
+	GraphContext *gc = _DecodeHeader(rdb);
+		
+	// load the key schema
+	PayloadInfo *key_schema = _RdbLoadKeySchema(rdb);
+
+	// the decode process contains the decode operation of many meta keys
+	// representing independent parts of the graph
+	//
+	// each key contains data on one or more of the following:
+	// 1. nodes - The nodes that are currently valid in the graph
+	//
+	// 2. deleted nodes - Nodes that were deleted and there ids can be re-used
+	//    used for exact replication of data block state
+	//
+	// 3. edges - The edges that are currently valid in the graph
+	//
+	// 4. deleted edges - edges that were deleted and there ids can be re-used
+	//    used for exact replication of data block state
+	//
+	// 5. graph schema - properties, indices
+	//
+	// the following switch checks which part of the graph the current key holds
+	// and decodes it accordingly
+	uint payloads_count = array_len(key_schema);
+	for(uint i = 0; i < payloads_count; i++) {
+		PayloadInfo payload = key_schema[i];
+		switch(payload.state) {
+			case ENCODE_STATE_NODES:
+				Graph_SetMatrixPolicy(gc->g, SYNC_POLICY_NOP);
+				RdbLoadNodes_v10(rdb, gc, payload.entities_count);
+				break;
+			case ENCODE_STATE_DELETED_NODES:
+				RdbLoadDeletedNodes_v10(rdb, gc, payload.entities_count);
+				break;
+			case ENCODE_STATE_EDGES:
+				Graph_SetMatrixPolicy(gc->g, SYNC_POLICY_NOP);
+				RdbLoadEdges_v10(rdb, gc, payload.entities_count);
+				break;
+			case ENCODE_STATE_DELETED_EDGES:
+				RdbLoadDeletedEdges_v10(rdb, gc, payload.entities_count);
+				break;
+			case ENCODE_STATE_GRAPH_SCHEMA:
+				// skip, handled in _DecodeHeader
+				break;
+			default:
+				ASSERT(false && "Unknown encoding");
+				break;
+		}
+	}
+	array_free(key_schema);
+
+	// update decode context
+	GraphDecodeContext_IncreaseProcessedKeyCount(gc->decoding_context);
+
+	// before finalizing keep encountered meta keys names, for future deletion
+	const RedisModuleString *rm_key_name = RedisModule_GetKeyNameFromIO(rdb);
+	const char *key_name = RedisModule_StringPtrLen(rm_key_name, NULL);
+	// the virtual key name is not equal the graph name
+	if(strcmp(key_name, gc->graph_name) != 0) {
+		GraphDecodeContext_AddMetaKey(gc->decoding_context, key_name);
+	}
+
+	if(GraphDecodeContext_Finished(gc->decoding_context)) {
+		Graph *g = gc->g;
+
+		// revert to default synchronization behavior
+		Graph_SetMatrixPolicy(g, SYNC_POLICY_FLUSH_RESIZE);
+		Graph_ApplyAllPending(g, true);
+
+		// QueryCtx_SetGraphCtx(gc);
+		
+		uint label_count = Graph_LabelTypeCount(g);
+		// update the node statistics
+		for(uint i = 0; i < label_count; i++) {
+			GrB_Index nvals;
+			RG_Matrix L = Graph_GetLabelMatrix(g, i);
+			RG_Matrix_nvals(&nvals, L);
+			GraphStatistics_IncNodeCount(&g->stats, i, nvals);
+		}
+
+		// make sure graph doesn't contains may pending changes
+		ASSERT(Graph_Pending(g) == false);
+
+		QueryCtx_Free(); // release thread-local variables
```

Comment:
now that `QueryCtx_SetGraphCtx` is commented out, see if we can remove the call to `GraphDecodeContext_Reset` ?

Reply:
we can't remove it it is set also in the header

Reply:
Call `QueryCtx_Free` at the end right before `return gc` apply to all decoders.